grafast/grafserv/src/servers/deno/v1/index.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  convertHandlerResultToResult,
+  GrafservBase,
+  normalizeRequest,
+} from "grafserv";
+
+import type {
+  GrafservBodyJSON,
+  GrafservConfig,
+  RequestDigest,
+  Result,
+} from "grafserv";
+
+declare global {
+  namespace Grafast {
+    interface RequestContext {
+      denov1: {
+        ctx: Deno.ServeHandlerInfo;
+      };
+    }
+  }
+}
+
+function getDigest(req: Request, ctx: Deno.ServeHandlerInfo): RequestDigest {
+  const url = new URL(req.url);
+  return {
+    // TODO: figure out the actual HTTP version
+    httpVersionMajor: 2,
+    httpVersionMinor: 0,
+    isSecure: url.protocol === "https",
+    method: req.method,
+    path: url.pathname,
+    headers: Object.fromEntries(req.headers.entries()),
+    getQueryParams() {
+      // TODO: on duplicate keys, this should convert to an array instead
+      return Object.fromEntries(url.searchParams) as Record<
+        string,
+        string | string[]
+      >;
+    },
+    async getBody() {
+      return {
+        type: "json",
+        json: await req.json(),
+      } as GrafservBodyJSON;
+    },
+    requestContext: {
+      denov1: {
+        ctx,
+      },
+    },
+  };
+}
+
+export class DenoGrafserv extends GrafservBase {
+  constructor(config: GrafservConfig) {
+    super(config);
+  }
+  public async handler(req: Request, ctx: Deno.ServeHandlerInfo) {
```

Comment:
The constructor wasn't needed.

```suggestion
  public async handler(req: Request, ctx: Deno.ServeHandlerInfo) {
```